securecookie.go

```diff
@@ -205,27 +206,32 @@ func checkPath(path string) error {
 // See https://tools.ietf.org/html/rfc1034#section-3.5 and
 // https://tools.ietf.org/html/rfc1123#section-2.
 func checkDomain(name string) error {
-	if len(name) > 255 {
+	switch {
+	case len(name) == 0:
+		return nil // an empty domain name will result in a cookie without a domain restriction
+	case len(name) > 255:
```

Comment:
A label is also restricted to be at most 63 byte long. This is currently unchecked.